qt/widgets/common/test/FitPropertyBrowserTest.h

```diff
@@ -94,6 +95,45 @@ class FitPropertyBrowserTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(oldExpString.toStdString(), expectedTieExpression);
   }
 
+  void test_removeFunctionRemovesTie() {
+    m_fitPropertyBrowser->init();
+    m_fitPropertyBrowser->createCompositeFunction(
+        "name=Gaussian,Height=10.0,PeakCentre=-0.145,Sigma=0.135;name=Gaussian,Height=12.0,PeakCentre=0.245,Sigma=0."
+        "135;ties=(f0.Height=f1.Height)");
+    auto f0Handler = m_fitPropertyBrowser->getPeakHandler(QString("f0"));
+    auto f1Handler = m_fitPropertyBrowser->getPeakHandler(QString("f1"));
+    TS_ASSERT(f0Handler->hasTies());
+    m_fitPropertyBrowser->removeFunction(f1Handler);
+    TS_ASSERT(!f0Handler->hasTies());
+  }
+
+  void test_removeFunctionRenamesOtherFunction() {
+    m_fitPropertyBrowser->init();
+    m_fitPropertyBrowser->createCompositeFunction(
+        "name=Gaussian,Height=10.0,PeakCentre=-0.145,Sigma=0.135;name=FlatBackground,A0=10;"
+        "ties=(f0.Height=f1.A0)");
+    auto f0Handler = m_fitPropertyBrowser->getPeakHandler(QString("f0"));
+    TS_ASSERT(QString("f0-Gaussian") == f0Handler->functionName());
+    m_fitPropertyBrowser->removeFunction(f0Handler);
+    // f0 should now be the flat background function
+    f0Handler = m_fitPropertyBrowser->getPeakHandler(QString("f0"));
+    TS_ASSERT(!f0Handler->hasTies());
+    TS_ASSERT_EQUALS(QString("f0-FlatBackground"), f0Handler->functionName());
+  }
+
+  void test_removeFunctionUpdatesTieString() {
+    m_fitPropertyBrowser->init();
+    m_fitPropertyBrowser->createCompositeFunction(
+        "name=Gaussian,Height=10.0,PeakCentre=-0.145,Sigma=0.135;name=FlatBackground,A0=10;"
+        "name=Gaussian,Height=10.0,PeakCentre=-0.555,Sigma=0.135;ties=(f0.Height=f2.Height)");
+    auto f0Handler = m_fitPropertyBrowser->getPeakHandler(QString("f0"));
+    auto f1Handler = m_fitPropertyBrowser->getPeakHandler(QString("f1"));
+    TS_ASSERT_EQUALS(m_fitPropertyBrowser->getOldExpressionAsString(QString("f0.Height")), QString("f2.Height"));
+    m_fitPropertyBrowser->removeFunction(f1Handler);
+    TS_ASSERT(f0Handler->hasTies());
+    TS_ASSERT_EQUALS(m_fitPropertyBrowser->getOldExpressionAsString(QString("f0.Height")), QString("f1.Height"));
```

Comment:
`getOldExpressionAsString` is a rather worrying name!
https://github.com/mantidproject/mantid/blob/ec825fbd8c2d754b56329d9056e8016ac6d7b10a/qt/widgets/common/src/FitPropertyBrowser.cpp#L1462-L1464
The comment seems to imply this only works with tied parameters with the same name across functions.
Is it possible to test this for a general tie (e.g. `f0.Height` tied to `f1.Sigma`)?

Reply:
Okay, I can try that. I can try and avoid that function if you like. It shouldn't be needed really.